block/manager.go

```diff
@@ -603,6 +603,10 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 		block = m.createBlock(newHeight, lastCommit, lastHeaderHash)
 		m.logger.Debug("block info", "num_tx", len(block.Data.Txs))
 
+		block.SignedHeader.DataHash, err = block.Data.Hash()
```

Comment:
at this point you don't have the ISRs, so why need DataHash here? it will anyway be computed afterwards (outside of if-else)?
i recommend refactoring the code to do this once (datahash and getCommit) for both pendingBlock and fresh block.

Reply:
You need to recompute `DataHash` here in order to pass the block validation logic inside `applyBlock`. If I remove it as you did in this PR: https://github.com/rollkit/rollkit/pull/858/files#diff-f64dae91a36980ba02c538f69fb83b7c80a8b4345db96849127307d33ccfacd1L140-L144 then we would be calling `applyBlock` for ISR computation without ANY block validation which is not something we'd like to avoid.

Maybe in the future, ideally, we separate out the validation logic so that `DataHash` validation and other validation logic are separate. In that case, yes, we can not set `DataHash` after `createBlock` and replace the validaiton logic in `applyBlock` with the other validation logic.

Reply:
not sure i understand the argument here. i think we can move the datahash computation and getCommit which is right after the createBlock to outside and do it only once instead of twice which is unnecessary. 
in my old PR, i did call `Validate` before applyBlock: https://github.com/rollkit/rollkit/pull/858/files#diff-0a427fe6ee00070e86bdb14930af6ae758cf682d702e12a2f386c56f20a5de2bL328

Reply:
my bad. it make sense.